src/stubs/sm_exit.s

```diff
@@ -56,11 +58,14 @@ __reti_entry:
     pop r13
     pop r14
     pop r15
+    pop r2
```

Comment:
careful, moving here in R2 also re-enables IRQs, but the SP and R1 have not yet been restored so might be lost right?!

Reply:
I see two options here:
1) ignore it as we plan to change the behavior of r2 anyway. Then, an eint in r2 would not have an effect anyway
2) Pop r2 into a different register, remove the gint bit and move it into r2

Reply:
btw this is resolved by now with a mix: 1) GIE bit is cleared before the pop r2 line and writes to IRQ bit are ignored by the atomicity monitor anyway if it is not the SM ID 1 running.